hooks/boot/20-certs_update.rb

```diff
@@ -18,6 +18,14 @@
   :default => false
 )
 app_option(
+  '--certs-reset',
+  :flag,
+  "This option will reset any custom certificates and use the self-signed CA " \
+  "instead. Note that any clients will need to be updated with the latest " \
+  "katello-ca-consumer RPM.",
```

Comment:
Pfft, you and your questions!  I think (1) and (2) should be OK, but l'll actually test with a proxy to make sure.